parser-typechecker/src/Unison/Codebase/ShortBranchHash.hs

```diff
@@ -1,15 +1,28 @@
-{- ORMOLU_DISABLE -} -- Remove this when the file is ready to be auto-formatted
-module Unison.Codebase.ShortBranchHash where
+module Unison.Codebase.ShortBranchHash
+  ( prefixName,
+    toString,
+    toHash,
+    fromHash,
+    fullFromHash,
+    fromText,
+    ShortBranchHash (..),
+  )
+where
 
-import Unison.Prelude
-import qualified Unison.Hash as Hash
-import qualified Data.Text as Text
 import qualified Data.Set as Set
+import qualified Data.Text as Text
+import qualified Unison.Hash as Hash
+import Unison.Name (Name)
+import qualified Unison.Name as Name
+import Unison.NameSegment (NameSegment (NameSegment))
+import Unison.Prelude
 
-newtype ShortBranchHash =
-  ShortBranchHash { toText :: Text } -- base32hex characters
+newtype ShortBranchHash = ShortBranchHash {toText :: Text} -- base32hex characters
   deriving stock (Eq, Ord, Generic)
 
+prefixName :: ShortBranchHash -> Name -> Name
```

Comment:
I'm reluctant to introduce a dependency on `Unison.Name` in this module. Also, not sure how this is used yet, but if `toText sbh` starts with a digit, which it can, then we don't have a way to parse that `NameSegment` in the CLI currently.

Reply:
This is analogous to `Path.prefixName` just as a named Path id analogous to a SBH in that they both identify a branch.

Is there some other spot you'd prefer I put this? Is there some other way you'd prefer to implement it?

I suppose it comes down to whether `#hkrqt3tm05.base.List` should be a valid name or not.

I think it should be, how else could I refer to names on a history branch or orphaned root?

As far as starting with a digit, good catch, didn't realize `ShortBranchHash.toText` behaved differently to `ShortHash.toText`; I think that's a bit of a footgun so I changed `SBH.toText` to behave the same, adjusted any usages, and added `SBH.toBase32Hex` to avoid any confusion.